app/models/supplejack_api/user.rb

```diff
@@ -64,6 +65,12 @@ def custom_find(id)
       end
     end
 
+    def update_user_sets
+      return unless username_changed?
+
+      user_sets.each { |set| set.update_attribute(:username, username) }
```

Comment:
can you a simpler `user_sets.update_all(username: username)` ? It wouldn't trigger the callbacks on each set, but I think that's fine.